### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by ysk24ok**
* Update documents for `pfi::data::unordered_map` and `pfi::lang::shared_ptr` .
* Add a document for `pfi::data::unordered_set`
* Suppress warnings when building docs
* Update gcc version in README.rst and README.ja.rst

### CODE REVIEW

File: doc/source/network/cgi.rst [hunk 1]
```diff
@@ -144,7 +144,7 @@ run_cgi、run_fcgiの代わりにrun_serverを用いると、
 
 argcとargvを渡すコンストラクタを用いると、コマンドライン引数でサーバの設定をすることができる。
 
-.. code-block:: c++
+.. code-block:: bash
```
Comment (ysk24ok) [inline doc/source/network/cgi.rst:147]:
```suggestion
.. code-block:: console
```

**Review by ysk24ok — COMMENTED** [review]
- Update doc/source/about.rst
  - Delete「また、C++0xおよび新しめのコンパイラが利用できない環境も想定している。」「tr1や」
- Update doc/source/install.rst
  - Change Python version to 3.6 and 3.7.
  - Change gcc version to 4.8.5 - 9.3.0
  - Delete Magick++

**Comment by ysk24ok** [comment]
Fixed problems stated in https://github.com/retrieva/pficommon/pull/215#pullrequestreview-510188432 by https://github.com/retrieva/pficommon/pull/215/commits/5469481035dbf332ac421bfdbae64838762b6e3f .

**Comment by takei-yuya** [comment]
LGTM!!

**Pull request merged**